src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -299,24 +299,66 @@ namespace OpenLoco::World::MapGenerator
         applySurfaceStyleToMarkedTiles(heightMap, surfaceStyle, false);
     }
 
+    // 0x0046A0D8, 0x0046A227
+    static void generateTerrainInRandomAreas([[maybe_unused]] HeightMap& heightMap, uint8_t surfaceStyle, uint16_t randAmount, uint16_t baseAmount)
+    {
+        auto ecx = getGameState().rng.randNext(64) + 0x50;
+        for (; ecx > 0; ecx--)
+        {
+            // TODO: could probably simplify / replace with two randNext(lo, hi) calls
+            auto randPos = getGameState().rng.randNext();
+            auto xPos = (((randPos & 0xFFFF) * kMapColumns) >> 16) & 0xFF;
```

Comment:
pretty sure you shouldn't be & 0xFF these. Thats going to remove the upper 1 bit (remember we have 384 not 255 limit)

Reply:
I was doing so because it operated on ax and cx instead of eax and ecx:
```asm
shl     ax, 5
shl     cx, 5
```

But I realise now they're 16-bit registers, not 8-bit, so I thought wrong. I'll remove the & 0xFF.